packages/lodestar-beacon-state-transition/src/fast/util/epochContext.ts

```diff
@@ -49,7 +49,8 @@ export class EpochContext {
   public currentShuffling!: IEpochShuffling;
   public nextShuffling!: IEpochShuffling;
   public config: IBeaconConfig;
-  public epochProcess?: IEpochProcess;
+  // validator balance at the time of epoch transition
+  public epochBalances?: Gwei[];
```

Comment:
I think we should expose the epochProcess to lodestar like:
```typescript
interface IStateContext {
  state: BeaconState;
  epochCtx: EpochContext;
  epochProcess?: EpochProcess;
}
```
And then downstream, in lodestar, remove the epoch process / store balances:
```typescript
...
// remove the epoch process
const ctx = {
  state: stateCtx.state,
  epochCtx: stateCtx.epochCtx,
};
// pull out the balances
const balances = stateCtx.epochProcess.statuses.map((s) => (s.active ? s.validator.effectiveBalance : BigInt(0)));

// store trimmed down state ctx
await db.stateCache.add(ctx);
await db.checkpointStateCache.add(checkpoint, ctx);
// store balances
await db.checkpointBalanceCache.add(checkpoint, balances);
```

Reply:
so we introduce a new `checkpointBalanceCache` in db? Do we have any other use cases in the future using this?
also, I suppose we should have `prune` logic similar to what we have with `checkpointStateCache`

fyi I still got OOM using this PR (I thought we don't have to reduce `MAX_EPOCHS`), I'll try again with `MAX_EPOCH` as 10. The memory usage should be the same with this change.

Reply:
> so we introduce a new checkpointBalanceCache in db? Do we have any other use cases in the future using this?

Yea this is just a first idea. What I don't like about the current situation is adding ad-hoc pieces of the epochProcess to the epochCtx. Trying to avoid that.
There's also an issue if we don't have the balances (or epochCtx) in the cache.

Perhaps we can save all of the cache changes for a separate PR.
Simply have this PR do: "Use balances of justified checkpoint state instead of current checkpoint for forkchoice".